### PR TITLE
feat: TxDAG generation v0.1 version

### PR DESCRIPTION
**Pull request by welkin22**
### Description

Mainly includes the following contents:
1. [Add the dependency of read-before-write to the DAG](https://github.com/bnb-chain/op-geth/pull/187/commits/7989ace3ba4f5c79a51d143486ff0acfe686b830)
2. [Remove redundant data when generating NonDependentRelFlag](https://github.com/bnb-chain/op-geth/pull/187/commits/badc36cc9c77911d2556db04d9cb328fbdd18c53)
3. All the content of https://github.com/bnb-chain/op-geth/pull/149

### Rationale

Version 0.1 of the TxDAG generation code is based on https://github.com/bnb-chain/op-geth/pull/149 with some minor modifications.

### Example

Add two parameters to geth:
```
--parallel.txdag
--parallel.txdagsenderpriv=xxx
```
`--parallel.txdag` will enable the TxDAG generation feature, while `parallel.txdagsenderpriv` specifies the private key of the sender for the transaction into which the generated TxDAG data is inserted.

### CODE REVIEW

**Review by sunny2022da — COMMENTED** [review]
LGTM, still need review from others

**Review by redhdx — DISMISSED** [review]

File: miner/worker.go [hunk 1]
```diff
@@ -1036,6 +1045,87 @@ func (w *worker) commitTransactions(env *environment, plainTxs, blobTxs *transac
...
+func (w *worker) generateDAGTx(statedb *state.StateDB, signer types.Signer, txIndex int, gasLimitForDag uint64) (*types.Transaction, error) {
+	if statedb == nil {
+		return nil, fmt.Errorf("failed to get state db, env.state=nil")
+	}
+
+	if signer == nil {
+		return nil, fmt.Errorf("current signer is nil")
+	}
+
+	//privateKey, err := crypto.HexToECDSA(privateKeyHex)
```
Comment (owen-reorg) [inline miner/worker.go:1079]:
```suggestion
```

File: tests/block_test.go [hunk 2]
```diff
@@ -73,20 +73,72 @@ func TestExecutionSpecBlocktests(t *testing.T) {
 	})
 }
 
+func TestBlockchainWithTxDAG(t *testing.T) {
+	//log.SetDefault(log.NewLogger(log.NewTerminalHandlerWithLevel(os.Stderr, log.LevelDebug, true)))
```
Comment (owen-reorg) [inline tests/block_test.go:77]: plz remove useless code in this test

**Review by sunny2022da — APPROVED** [review]
LGTM

**Review by owen-reorg — APPROVED** [review]

**Pull request merged**